packages/react-composites/src/composites/CallWithChatComposite/CustomButton.tsx

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { ControlBarButton } from '@internal/react-components';
+import React from 'react';
+import { generateCustomControlBarButtonStrings } from '../CallComposite/components/buttons/Custom';
+import {
+  CallControlDisplayType,
+  CustomCallControlButtonCallbackArgs,
+  CustomControlButtonProps
+} from '../CallComposite/types/CallControlOptions';
+import { CallWithChatCompositeIcon, CallWithChatCompositeIcons } from '../common/icons';
+/* @conditional-compile-remove(control-bar-button-injection) */
+import { CallWithChatControlOptions } from './CallWithChatComposite';
+
+/**
+ * Max number of Custom Buttons in primary and secondary ControlBar
+ * Does not include existing buttons in the controlBar.
+ *
+ * @private
+ */
+export const CUSTOM_BUTTON_OPTIONS = {
+  MAX_PRIMARY_DESKTOP_CUSTOM_BUTTONS: 3,
+  MAX_PRIMARY_MOBILE_CUSTOM_BUTTONS: 1,
```

Comment:
How do we decide on the cut off here? Only a very narrow device I don't think we have any room for a custom button

Reply:
I think I'm more in favor of not trying to be smart and letting contoso decide where they put the buttons - they can detect what device size they are running on and put buttons in the primary or overflow accordingly

Reply:
The MAX custom buttons for each platform/screen was provided by Alex P. 
But good point on "What happens when Contoso decides to hide buttons" in your other comment. 

I am okay with not limiting how many buttons get injected. However, this would complicate how we "automagically" move buttons into the overflow menu.

Reply:
Yeah the automagically part is tough... e.g. Do we do it on container size or based on a fixed number? If we do a fixed number what if the user knows their site will only be on larger devices and know they have more screen real estate to make use of.. 
I won't block this PR on the automagical moving of buttons, we can always update later, but I think it's complicated enough with enough edge cases to warrant its own section in your proprosed solution 4 to call out how this should be implemented

Reply:
Container size vs fixed number, I can see the argument for both.

The idea with max custom buttons was "Guaranteeing to Contoso that X # of Custom Buttons would be injected depending on desktop or mobile". 

However, implementing based on container size would be ideal but would require quite a bit of change. I like this approach and will explore this avenue. Will also discuss further with Alex P. on design if this is to be the case. 
I will also add onto Proposal #4, automagically moving buttons warrants clarity and further outlining.

For now, should we remove max buttons? (this would remove automagically moving buttons into the Overflow from this PR)